ksql-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java

```diff
@@ -124,12 +120,17 @@ public Field getKeyField() {
 
     final KsqlStructuredDataOutputNode noRowKey = outputNodeBuilder.build();
     if (doCreateInto) {
+      final SourceTopicProperties sourceTopicProperties = getSourceTopicProperties(
```

Comment:
@rodesai suggests that we disable the check on the partition and replica count for pre-existing sink topic in this comment: https://github.com/confluentinc/ksql/pull/1930#discussion_r220069905.

I can't see that you've done that, but may be missing something - have you done it?

If we have/are doing it, then my original comment about making the lookup of the source topics partition and replica count still stands.  If the sink topic already exists, and we're not validating partition and replica counts, then we do not need to get the source topics info, so we shouldn't as its a cross-machine call.

Reply:
OK, confused, as you start in the comment that I linked to that:

> I updated it as we talked offline.

Maybe for the benefit of others reviewing you can explain what was discussed and what was updated?

Reply:
We check the source topic properties if they are not provided by the user in the WITH clause. We also have to check the sink topic properties to make sure they conform for the properties that we require. If there is an existing topic with partitions or replicas other than the ones we require we will fail.